### PR TITLE
Add editorconfig

### PR DESCRIPTION
**Pull request by SimenB**
It silences IntelliJ a least 😄 
I think this should match the style in the repo.

image

### CODE REVIEW

**Comment by ljharb** [comment]
I don't think it's a good idea to include this file unless we can validate it against our existing code. Do you know of a tool that can do that?

**Comment by SimenB** [comment]
Well, there is https://github.com/jedmao/eclint. I'm unable to make it work properly, though...

But the files use 2 space indent, trailing newline, and no trailing spaces.

All of these are validated by eslint already (maybe not utf-8, but that's usually default in modern editors, and I doubt this repo uses anything other than ascii either way), this file just automatically sets the IDE/text editor to those same settings.

**Comment by Vadorequest** [comment]
This may be what you need, I haven't looked deeply into it but it seems good.

https://github.com/sun1x/eslint-to-editorconfig

**Comment by SimenB** [comment]
@Vadorequest so the test would be to run the tool, and expect `git status` to be clean?

**Comment by Vadorequest** [comment]
I think I misunderstood what you wanted, I was looking for a tool to generate a .editorconfig based on ESLint. 

But yeah, you could generate the file using this module and match it against yours.

**Comment by SimenB** [comment]
@ljharb Found a tool to run a check on CI (same that webpack uses)

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+indent_size = 2
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+end_of_line = lf
+# editorconfig-tools is unable to ignore longs strings or urls
+max_line_length = null
+
+[*.md]
+trim_trailing_whitespace = false
```
Comment (ljharb) [inline .editorconfig]: why would we not want to trim trailing whitespace on markdown files?

File: packages/eslint-config-airbnb-base/package.json [hunk 2]
```diff
@@ -4,6 +4,7 @@
   "description": "Airbnb's base JS ESLint config, following our styleguide",
   "main": "index.js",
   "scripts": {
+    "prelint": "editorconfig-tools check *.js rules/*.js test/*.js",
```
Comment (ljharb) [inline packages/eslint-config-airbnb-base/package.json]: huh, i wasn't aware you could "pre" any old script. Seems this works all the way back to npm 1.

Comment (ljharb) [inline packages/eslint-config-airbnb-base/package.json]: This should probably cover jsx files too.

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,14 @@
+root = true
+
+[*]
```
Comment (ljharb) [inline .editorconfig]: if this is `*`, then the editorconfig check should also use `*` - if it's just js/jsx, then i'd like both places to match.

Comment (SimenB) [inline .editorconfig]: that's a good point

File: .editorconfig (hunk 1 above)
Comment (SimenB) [inline .editorconfig]: MD has significant whitespace, so if you want e.g. a linebreak without starting a new paragraph, you add two trailing spaces.

image

Want me to remove it?

**Comment by SimenB** [comment]
Changed to run against every file

File: .editorconfig (hunk 1 above)
Comment (ljharb) [inline .editorconfig]: Interesting, I was not aware of that.

It's fine to leave as-is then.

**Comment by SimenB** [comment]
Yes, it walks up the tree. I know it worked because it gave me failures on too long lines before I disabled that rule

**Comment by SimenB** [comment]
I don't think the file needs to bi included in a publish, it's just useful for contributors, having their IDE configured automatically

**Comment by ljharb** [comment]
@SimenB `npm explore foo && npm install && npm test` should work for every package - it definitely has to be included in the publish.

**Comment by SimenB** [comment]
hm, ok. add a symbolic link, duplicate it, or copy it over during pubilsh (and gitignore it)?

**Comment by ljharb** [comment]
I think the symlink is probably the simplest solution.

**Comment by SimenB** [comment]
Ok, symlinks pushed

**Comment by SimenB** [comment]
@ljharb rebased

**Comment by ljharb** [comment]
hmm - when i try this locally, even if i add a hard tab, it doesn't error out.

**Comment by SimenB** [comment]
Hmm, it doesn't seem to be able to check indent style... In the source code it's called `indent-char`, but changing to that doesn't help.

It does correctly yell at me for trailing spaces, trailing newline, CRLF and size of indentation, though. And it also yelled at me for line length, but as it\s unable to ignore long strings or comments, it gave a lot of false positives.

A bummer it doesn't seems to check type of indent, but extra indent gets yelled at

**Comment by SimenB** [comment]
Also, setting `indent-style=tab` makes it fail for spaces, so it just doesn't seem able to detect a single tab in all the spaces

**Comment by ljharb** [comment]
gotcha - seems i just got lucky and picked the one thing it can't detect. oh well, this seems fine :-)

**Comment by SimenB** [comment]
Hmm, If I set _all_ indentation to tabs, it yells at me for not using spaces... Seems like a bug

**Comment by ljharb** [comment]
@SimenB please file that upstream if you don't mind ;-)

**Comment by SimenB** [comment]
Done, slang800/editorconfig-tools#36

Thanks for merging!

**Pull request merged**